src/extensions/renderer/canvas/drawing-edges.js

```diff
@@ -208,8 +208,18 @@ CRp.drawEdgePath = function( edge, context, pts, type ){
       case 'straight':
       case 'segments':
       case 'haystack':
-        for( let i = 2; i + 1 < pts.length; i += 2 ){
-          context.lineTo( pts[ i ], pts[ i + 1] );
+        const taxiRadius= edge.pstyle( 'taxi-radius' )
+        const taxiRadiusValue = taxiRadius ? taxiRadius.value : 0
+
+        if( taxiRadiusValue > 0) {
```

Comment:
This block should only apply to taxi edges, or the property should explicitly support all the affected cases.  

Currently, it's breaking other edges.  E.g. on the debug page with all edges set with `taxi-radius: 32px`:

Expected:

<img width="408" alt="Screenshot 2023-08-01 at 08 46 33" src="https://github.com/cytoscape/cytoscape.js/assets/989043/053540bb-0128-499d-8596-83dcdd107f15">

Actual:

<img width="450" alt="Screenshot 2023-08-01 at 08 46 18" src="https://github.com/cytoscape/cytoscape.js/assets/989043/057f106d-89f6-4971-9d80-3b60027b01db">